rlclientlib/model_mgmt/restapi_data_transport.h

```diff
@@ -3,23 +3,30 @@
 #include "utility/http_client.h"
 
 #include <chrono>
+#include <cpprest/http_headers.h>
 #include <string>
 
+using namespace web::http;
+
 namespace reinforcement_learning {
   class i_trace;
   namespace model_management {
   class restapi_data_transport : public i_data_transport {
   public:
     // Takes the ownership of the i_http_client and delete it at the end of lifetime
     restapi_data_transport(i_http_client* httpcli, i_trace* trace);
+    restapi_data_transport(i_http_client* httpcli, http_headers& header, bool api, i_trace* trace);
 
     int get_data(model_data& data, api_status* status) override;
   private:
     using time_t = std::chrono::time_point<std::chrono::system_clock>;
-    int get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status);
+    int get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status, http_request request);
```

Comment:
I see http_request is already saved as a member. Do we still need it in the signature?

Reply:
member is not requirer. we need to pass in signature as depending on the type of URI we get we need to change request